bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java

```diff
@@ -0,0 +1,154 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.mob41.blapi.RM2Device;
+import com.github.mob41.blapi.mac.Mac;
+import com.github.mob41.blapi.pkt.cmd.rm2.SendDataCmdPayload;
+
+/**
+ * The {@link RMUniversalRemoteHandler} is responsible for handling RM Mini Universal Remotes.
+ *
+ * @author Miguel Álvarez - Initial contribution
+ */
+@NonNullByDefault
+public class RMUniversalRemoteHandler extends BroadlinkBaseHandler {
+    private final Logger logger = LoggerFactory.getLogger(RMUniversalRemoteHandler.class);
+    private @Nullable RM2Device rm2Device;
+
+    /**
+     * Creates a new instance of this class for the {@link RMUniversalRemoteHandler}.
+     *
+     * @param thing the thing that should be handled, not null
+     */
+    public RMUniversalRemoteHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Initializes a new instance of a {@link RMUniversalRemoteHandler}.
+     */
+    @Override
+    public void initialize() {
+        super.initialize();
+        if (host != null && macAddress != null) {
+            try {
+                blDevice = new RM2Device(host, new Mac(macAddress));
+                this.rm2Device = (RM2Device) blDevice;
+                updateStatus(ThingStatus.ONLINE);
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Could not find broadlink universal remote device at host" + host + "with MAC+" + macAddress
+                                + ": " + e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command: {}", command.toFullString());
+        authenticate(false);
+
+        if (command == RefreshType.REFRESH) {
+            logger.warn("Nothing to refresh on thing {}", thing.getUID());
+            return;
+        }
+        String channelId = channelUID.getIdWithoutGroup();
+        try {
+            switch (channelId) {
+                case BroadlinkBindingConstants.LEARNING_MODE:
+                    handleLearningCommand();
+                    break;
+                case BroadlinkBindingConstants.SAVE_LEARNED:
+                    handleSaveLearned(command);
+                    break;
+                case BroadlinkBindingConstants.SEND_LEARNED:
+                    handleSendLearned(command);
+                    break;
+                default:
+                    logger.warn("Channel {} does not supported by thing {}", channelId, thing.getUID());
+            }
+        } catch (Exception e) {
+            logger.warn("IOException while running channel {}", channelUID);
+        }
+    }
+
+    private void handleLearningCommand() throws IOException {
+        RM2Device rm2Device = this.rm2Device;
+        if (rm2Device != null) {
+            try {
+                rm2Device.enterLearning();
```

Comment:
There is no way to exit properly from learning mode ?

Reply:
It's exited when check data method is called.
One option to implement it will be to call check data and ignore the result when and off command is send to startlearning.
Should I added it?

Reply:
I let you decide.
At least now you added a check to handle only ON command, so it looks now better to me.

Reply:
I'll keep it like this then, I don't think it will be used.